webapp/src/main/java/com/box/l10n/mojito/pagerduty/PagerDutyClient.java

```diff
@@ -0,0 +1,113 @@
+import java.net.http.HttpResponse;
+import java.text.MessageFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * PagerDuty client for creating (triggering) and resolving incidents.
+ *
+ * @author mattwilshire
+ */
+public class PagerDutyClient {
+
+  static Logger logger = LoggerFactory.getLogger(PagerDutyClient.class);
+
+  static final String BASE_URL = "https://events.pagerduty.com";
+  static final String ENQUEUE_PATH = "/v2/enqueue";
+
+  public static final int MAX_RETRIES = 3;
+  public static String exceptionTemplate =
+      "PagerDuty request failed: Status Code: '{0}', Response Body: '{1}'";
+
+  private final HttpClient httpClient;
+  private final String integrationKey;
+
+  public PagerDutyClient(String integrationKey, HttpClient httpClient) {
+    if (integrationKey == null || integrationKey.isEmpty())
+      throw new IllegalArgumentException("Pager Duty integration key is null or empty.");
+    this.integrationKey = integrationKey;
+    this.httpClient = httpClient;
+  }
+
+  /**
+   * Trigger incident using a deduplication key and send PagerDutyPayload with it. The client will
+   * attempt to send the event request MAX_RETRIES times if an internal error is received from the
+   * server. If the max number of retries is reached, a bad request is received or the payload
+   * cannot be serialized a PagerDutyException is thrown.
+   */
+  public void triggerIncident(String dedupKey, PagerDutyPayload payload) throws PagerDutyException {
+    sendPayload(dedupKey, payload, PagerDutyPostData.EventAction.TRIGGER);
+  }
+
+  /**
+   * Resolve incident using a deduplication key. The client will attempt to send the event request
+   * MAX_RETRIES times if an internal error is received from the server. If the max number of
+   * retries is reached, a bad request is received or the payload cannot be serialized a
+   * PagerDutyException is thrown.
+   */
+  public void resolveIncident(String dedupKey) throws PagerDutyException {
+    sendPayload(dedupKey, null, PagerDutyPostData.EventAction.RESOLVE);
+  }
+
+  private void sendPayload(
+      String dedupKey, PagerDutyPayload payload, PagerDutyPostData.EventAction eventAction)
+      throws PagerDutyException {
+
+    if (dedupKey == null || dedupKey.isEmpty())
+      throw new PagerDutyException("Deduplication key should not be null or empty.");
+
+    PagerDutyPostData postBody = new PagerDutyPostData(integrationKey, dedupKey);
+    postBody.setPayload(payload);
+    postBody.setEventAction(eventAction);
+
+    try {
+      HttpRequest request = buildRequest(postBody.serialize());
+      sendRequest(request);
+    } catch (JsonProcessingException e) {
+      throw new PagerDutyException("Failed to serialize PagerDutyPostRequest to a JSON string.", e);
+    }
+  }
+
+  private void sendRequest(HttpRequest request) throws PagerDutyException {
+    // Retry sending PD notifications if they fail
+    for (int attempt = 1; attempt <= MAX_RETRIES; attempt++) {
```

Comment:
You can use the `Mono.retryable` methods from Project Reactor for the retry logic, it will allow you to configure retry strategies such as exponential backoff etc. from which you can expose configuration properties. 

You can see some examples of using this in the `SmartlingClient`

Reply:
Done, thanks for showing me this!